compiler-core/src/type_/hydrator.rs

```diff
@@ -76,7 +76,11 @@ impl Hydrator {
     /// an annotation. As such it should never be instantiated into an unbound
     /// variable.
     pub fn is_rigid(&self, id: &usize) -> bool {
-        self.rigid_type_ids.contains(id)
+        self.rigid_type_names.contains_key(id)
+    }
+
+    pub fn rigid_names(&self) -> im::HashMap<usize, String> {
+        self.rigid_type_names.clone()
```

Comment:
I think we could take ownership here, save the clone (even if it is a cheap one)
```suggestion
    pub fn take_rigid_names(&mut self) -> im::HashMap<usize, String> {
        std::mem::take(&mut self.rigid_type_names)
```

Reply:
Ah, I must have misread the code. I thought it was deallocated after that method was called, making it safe to remove that hashmap